tests/batch/test_AnalyserAndReporterDflowfm.py

```diff
@@ -160,17 +191,29 @@ def given_file_names_based_on_numbers_with_plotting_off_and_needs_tide_false_whe
 
         self.set_file_names()
 
-        mocked_mapfile = Mock(spec=MapFile)
+        face_node_connectivity = numpy.array([0, 1, 2, 3, 4])
+        mocked_mapfile = self._get_mocked_mapfile(face_node_connectivity)
+
+        xykm_data = self._get_mocked_xykm_data(self.xykm)
 
         with (
             patch("dfastmi.batch.ReporterDflowfm.MapFile", return_value=mocked_mapfile),
+            patch(
+                "dfastmi.batch.AnalyserDflowfm.OutputFileFactory.generate",
```

Comment:
are all these patches and mocking needed, doesn't the factory returning the mock not suffice?

Reply:
Yes, in ReporterDflowfm.py MapFile object is needed. Keep as is